src/requirements/Farcaster/hooks/useFarcasterCast.ts

```diff
@@ -0,0 +1,37 @@
+import useDebouncedState from "hooks/useDebouncedState"
+import useSWRImmutable from "swr/immutable"
+
+const BASE_URL =
+  "https://api.neynar.com/v2/farcaster/cast?api_key=NEYNAR_API_DOCS&identifier="
+
+const useFarcasterCast = (hash?: string, url?: string) => {
+  const debouncedHash = useDebouncedState(hash)
```

Comment:
Oh, one more comment, I think we should use the `useDebouncedState` hook inside the form component, so we don't use the debounced values in the requirement display component ;)